scripts/diagnostics.sh

```diff
@@ -1,145 +1,343 @@
+    while [[ "$#" -gt 0 ]]; do
+        case $1 in
+            -h|--help)
+                print_help; exit 0;;
+            -f|--full-build)
+                export FULL_BUILD=1; shift ;;
+            -r|--run-checks)
+                export RUN_CHECKS=1; shift ;;
+            -a|--active-environment)
+                export PRISTINE_ENVIRONMENT=0; shift ;;
+            -c|--conda-channel)
+                export CONDA_CHANNEL=$2; shift 2;;
+            *)
+                printf "Unknown argument: $1\n"; exit 1;;
+        esac
+    done
+    # default to printing active environment, otherwise do not print active environment
+    # (basically I want the default behavior to behave like the previous diagnostic.sh script)
+    if [ -n "${FULL_BUILD}" ] || [ -n "${RUN_CHECKS}" ]; then
+        NO_ENVIRONMENT=1
+    fi
+    # FULL_BUILD or RUN_CHECKS is set, but not PRISTINE_ENVIRONMENT
+    if [ "${PRISTINE_ENVIRONMENT}" != 0 ] && [ "${NO_ENVIRONMENT}" == 1 ]; then
+        printf "Beginning in a pristine environment\n"
+        env -i bash --rcfile <(echo "CLEAN_ENV='True' \
+                                PRISTINE_ENVIRONMENT=1 \
+                                METHODS=${METHODS:-opt} \
+                                METHOD=${METHOD:-opt} \
+                                MOOSE_JOBS=${MOOSE_JOBS:-6} \
+                                FULL_BUILD=${FULL_BUILD:-0} \
+                                RUN_CHECKS=${RUN_CHECKS:-0} \
+                                CONDA_CHANNEL=${CONDA_CHANNEL:-'https://conda.software.inl.gov/public'} \
+                                NO_ENVIRONMENT=1 \
+                                ${SCRIPT_DIR}/${BASH_SOURCE[0]}; exit")
+        exit $?
+    else
+        # Set defaults
+        export PRISTINE_ENVIRONMENT=${PRISTINE_ENVIRONMENT:-0}
+        export FULL_BUILD=${FULL_BUILD:-0}
+        export METHOD=${METHOD:-opt}
+        export METHODS=${METHODS:-opt}
+        export MOOSE_JOBS=${MOOSE_JOBS:-6}
+        export RUN_CHECKS=${RUN_CHECKS:-0}
+        export CONDA_CHANNEL=${CONDA_CHANNEL:-'https://conda.software.inl.gov/public'}
+        export NO_ENVIRONMENT=${NO_ENVIRONMENT:-0}
+    fi
+fi
+
+# Augment additional pristine environment variables
+if [ "${PRISTINE_ENVIRONMENT}" == "1" ]; then
+    # Make our environment more sane
+    export PATH=/bin:/usr/bin:/sbin
+    export TERM=xterm-256color
+    # Obtain our temp directory
+    export CTMP_DIR=$(create_tmp)
+    # Delete temporary directory at any time this script exits
+    trap 'rm -rf "$CTMP_DIR"' EXIT
+    printf '#%.0s' {1..60}
+    printf "\nTemporary Directory:\n${CTMP_DIR}\n"
+    printf '#%.0s' {1..60}
+    printf '\n\n'
+fi
+
+# Do only the one thing if default behavior
+if [ "${NO_ENVIRONMENT}" == 0 ]; then print_environment; exit; fi
+# Do all the things
+install_conda
+create_env
+download_moose
+printf "Using ${MOOSE_JOBS:-6} cores for \`make\` jobs.\n"
+build_petsc
+build_libmesh
+build_wasp
+build_moose
+test_moose
```

Comment:
this isnt diagnostics. this is fixing!

Reply:
haha, yeah. The whole reason I am doing this, is for those pesky machines (and HPC machines) which simply refuse to build MOOSE. I want a script I can ask a user to run, and it will eliminate their environment as a possible cause (or the clusters strange module loads).